docs/howto/customization/mapping_parser.md

```diff
@@ -1,12 +1,130 @@
 # How to write data to archive with MappingParser
 
-`MappingParser` is a generic parser class implemented in
-`nomad.parsing.file_parser/mapping_parser.py` to handle the conversion to and from a
+`MappingParser` is a NOMAD parsing framework for handling tree-like data structures.
+It is extensively used for processing highly structured files, like computational output.
+This framework distributes responsiblities between the file processing (building the source tree) and the mapping into the NOMAD schema.
+Some functionality can be inserted in the mapping, the rest is relegated to *normalization* functionalitites of the schema.
```

Comment:
I am not sure if we include the discussion of the infrastructure  here, i suggest you reference the already existing docs on parsing-normalization.